File: test_driver.jl
include("jump.jl")
include("exa.jl")

import JSON3

function test_driver()
    json_path = joinpath(@__DIR__, "test_cases.json")
    test_cases = open(json_path, "r") do f
        JSON3.read(json_path)
    end

    # get system args to select solver
    args = ARGS
    if length(args) == 0
        println("No solver selected. Defaulting to JuMP.")
        solver = "jump"
    else
        solver = args[1]
    end

    method = "polar"
    logdir = joinpath(@__DIR__, "log")
    if length(args) > 1 && args[2] == "rect"
        method = "rect"
        logdir = joinpath(@__DIR__, "log_rect")
    end

    if solver == "jump"
        println("Using JuMP solver.")
        f = jump_main
    elseif solver == "jump_symbolicad"
        println("Using JuMP SymbolicAD solver.")
        f = jump_symbolicad_main
    elseif solver == "ampl"
        println("Using AMPL solver.")
        f = ampl_main
    elseif solver == "exa"
        println("Using ExaModels solver.")
        f = exa_main
    else
        println("Invalid solver selected. Defaulting to JuMP.")
        exit(1)
    end

    for case in test_cases
        println("Running test case: $case")
        total_time, logpath = f(logdir, case, method)

        open(logpath, "a") do f
            write(f, "Total time: $total_time\n")
        end
    end
end

test_driver()

File: test_driver.py
from casadi_ import casadi_main
from poi import poi_main

import json
from pathlib import Path
import argparse

def main():
    json_path = Path(__file__).parent / "test_cases.json"

    with open(json_path, "r") as f:
        test_cases = json.load(f)

    # select solver method
    parser = argparse.ArgumentParser()
    parser.add_argument("--solver", type=str, default="poi")
    parser.add_argument("--method", type=str, default="polar")
    args = parser.parse_args()

    solver = args.solver
    method = args.method
    logdir = Path(__file__).parent / "log"

    if solver == "casadi":
        f = casadi_main
    elif solver == "poi":
        f = poi_main
    else:
        raise ValueError(f"Unknown solver: {solver}")
    
    if method == "rect":
        logdir = Path(__file__).parent / "log_rect"

    for case in test_cases:
        print(f"Running {case}")
        total_time, logpath = f(logdir, case, method)
        with open(logpath, "a") as file:
            file.write(f"Total time: {total_time}\n")

if __name__ == "__main__":
    main()
